components/api/src/main/java/com/hotels/styx/api/HttpInterceptor.java

```diff
@@ -57,6 +59,11 @@ public void add(String key, Object value) {
                 return null;
             }
         };
+
+        default <T> Optional<T> getOptional(String key, Class<T> clazz){
```

Comment:
I think we should use a name like `getIfAvailable` rather than `getOptional`, as it is better to describe the behaviour of the method rather than simply the return type, which is already visible.

Reply:
That's fine for me. I'll update w/ the change(and deprecation)